### PR TITLE
ocamlPackages.treewide: use dune 3 as default

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Depends on:
- https://github.com/NixOS/nixpkgs/pull/223745

Cherry-picks:
* 98dd9c8bf9e25fdb5a2fa61f25f567cd4a29da03

### CODE REVIEW

**Comment by Et7f3** [comment]
Why not add it in build support ?

**Comment by superherointj** [comment]
> I thought on deprecate useDune2 here

Removing `useDune2` is valid idea but out of scope for this PR.

**Comment by vbgl** [comment]
My opinion is that getting rid of `useDune2` is a prerequisite to making dune 3 the default.

File: pkgs/development/ocaml-modules/dune-configurator/default.nix [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 buildDunePackage rec {
   pname = "dune-configurator";
 
-  inherit (dune_2) src version patches;
+  inherit (dune_3) src version patches;
 
-  duneVersion = "2";
+  # This fixes finding csexp
+  postPatch = ''
+    rm -rf ./vendor
```
Comment (anmonteiro) [inline pkgs/development/ocaml-modules/dune-configurator/default.nix:10]: why are you removing the entire vendor dir, then some more?

**Comment by vbgl** [comment]
I think this can be closed.

**Pull request closed without merging**